server/events/vcs/azuredevops_client.go

```diff
@@ -57,33 +57,40 @@ func NewAzureDevopsClient(hostname string, userName string, token string) (*Azur
 // relative to the repo root, e.g. parent/child/file.txt.
 func (g *AzureDevopsClient) GetModifiedFiles(repo models.Repo, pull models.PullRequest) ([]string, error) {
 	var files []string
+	filesSet := make(map[string]bool)
 
-	opts := azuredevops.PullRequestGetOptions{
-		IncludeWorkItemRefs: true,
-	}
 	owner, project, repoName := SplitAzureDevopsRepoFullName(repo.FullName)
-	commitIDResponse, _, _ := g.Client.PullRequests.GetWithRepo(g.ctx, owner, project, repoName, pull.Num, &opts)
-
-	commitID := commitIDResponse.GetLastMergeSourceCommit().GetCommitID()
-
-	r, _, _ := g.Client.Git.GetChanges(g.ctx, owner, project, repoName, commitID)
 
-	for _, change := range r.Changes {
-		item := change.GetItem()
-		// Convert the path to a relative path from the repo's root.
-		relativePath := filepath.Clean("./" + item.GetPath())
-		files = append(files, relativePath)
-
-		// If the file was renamed, we'll want to run plan in the directory
-		// it was moved from as well.
-		changeType := azuredevops.Rename.String()
-		if change.ChangeType == &changeType {
-			// Convert the path to a relative path from the repo's root.
-			relativePath = filepath.Clean("./" + change.GetSourceServerItem())
-			files = append(files, relativePath)
+	commitRefs, _, _ := g.Client.PullRequests.ListCommits(g.ctx, owner, project, repoName, pull.Num)
+	_ = commitRefs
+	for _, ref := range commitRefs {
+		commitID := *ref.CommitID
+
+		r, _, _ := g.Client.Git.GetChanges(g.ctx, owner, project, repoName, commitID)
+
+		for _, change := range r.Changes {
+			item := change.GetItem()
+			if *item.GitObjectType == "blob" {
```

Comment:
Please check for a nil value prior to dereferencing `item`. L73 will cause a panic in the event that either `change` or `item` is nil. This library seems to have utility functions for accessing all of these properties which safely wraps the pointer dereferences for us. Lets use that.

```suggestion
			item := change.GetItem()
			if item.GetGitObjectType() == "blob" {
```